### DISCUSSION TITLE
Password reset

### DISCUSSION THREAD

**Discussion by danielmark0116** [TheWidlarzGroup/holidaily#26] [Ideas]
Opening a discussion on password reset

To sum up our discussion on Slack, we want to handle it with a 6 digits code received over an email after triggering the reset with a small form on a separate screen

User path:
- [ ] user clicks on `password reset` link
- [ ] user is redirected to a screen with `email` input and a `submit` button (+ proper validation)
- [ ] request is sent to the backend after submit
- [ ] based on whether the user exists, the app shows either a success or an error prompt of some sort (open topic for now :D)
- [ ] if successful, they also are redirected to the password reset confirm screen with an input for `code` and new `password` and a `submit` button
- [ ] if successful, user receives an email with 6 digits code (e.g. `4 2 6 8 5 9`)
- [ ] user types the code manually or can click a link below the code in the email which would be a universal/app link (DOMAIN.com/confirm_password_reset/OUR_6_DIGITS_CODE - such link would open the app automatically, prepopulating the aforementioned input automatically)

@domiskoczylas @iceu-bb @bemog @mateki0 @Nsquik @spaceowlpurrp @Okelm 

Let me know if I did not miss anything from yesterdays discussion

If we 100% agree on a given flow, we can create user stories out of it and start implementing it on the BE and front :)

**Comment by domiskoczylas** [comment]
User path:

- [ ] user clicks on password reset link
- [ ] user is redirected to a screen with email input and a submit button (+ proper validation)
- [ ] request is sent to the backend after submit
- [ ] the app shows either a success or an error
- [ ] if successful, the user goes to his inbox, and the app shows a screen for 6-digit code and a "paste" button
- [ ] if error, the user goes one step back, and re-types e-mail address
- [ ] the user copies 6 digit code from his inbox
- [ ] the user goes back to the app and pastes the 6 digit code with "paste" button
- [ ] If the code is correct, the next screen appears automatically - screen with reset password inputs and submit button.
- [ ] the user fills in the inputs with a new password (twice) and click submit button
- [ ] if successful, a dialog box appears, and inform that now the user can log in to his account


UI is ready [FIGMA](https://www.figma.com/file/LrAhsnwm7ZdUQPnKfjxfjz/Holidaily?node-id=0%3A1)
